simulator/func_sim/alu.h

```diff
@@ -268,8 +268,8 @@ struct ALU
     template<typename T> static void add_uw( Instr* instr) { instr->v_dst[0] = instr->v_src[1] + ( bitmask<T>(32) & instr->v_src[0]); }
   
     template<typename T> static void bclr  ( Instr* instr) { instr->v_dst[0] = instr->v_src[0] & ~( lsb_set<T>() << shamt_v_src2<T>( instr)); }
-
     template<typename T> static void bseti( Instr* instr)  { instr->v_dst[0] = instr->v_src[0] | (lsb_set<T>() << (shamt_imm (instr) & (XLEN - 1))); }
+    template<typename T> static void sext_b( Instr* instr) { instr->v_dst[0] = sign_extension<T>(instr->v_src[0], 8); }
```

Comment:
Please use CHAR_BITS or bitwidth<uint8> to indicate 8 stands for # bits in a byte